### PR TITLE
Refer to ESM modules, not CJS ones

### LINKED ISSUE #276: Try PureScript is broken
I deployed `v2022-06-08.1` and get the following erro:
> There was a problem making the request: AJAX request failed: POST https://compile.purescript.org/compile

Looking at the web dev's Network tab, I get the following message:
> Response body is not available to scripts (Reason: CORS Missing Allow Origin)

So, I redeployed `v2022-02-26.1`. However, the compiled code is no longer executed.

### PR DESCRIPTION
**Pull request by JordanMartinez**
**Description of the change**

Fixes #276.

#276 is caused by two issues:
- #275 changed the URL ultimately used to resolve the imports from https://compile.purescript.org/output to https://try.purescript.org/output. This approach meant we didn't need to remap the paths used in `import` statements as we could just use the `base` element. However, that means we send a request to the wrong domain.
- Originally, the modules served via `https://compile.purescript.org/output/*/.js` would return the files found in `/var/www/trypurescript/staging/.psci_modules/node_modules/$1`. This is no longer the place where such files are outputted in an ESM world (I'm not sure why). Rather, they're outputted in `/var/www/trypurescript/staging/.psci_modules/$1`. 

To summarize:
```diff
We need to fix the URL remapping approach to use the correct URL
-https://try.purescript.org/output
+https://compile.purescript.org/output

We need to update the alias for serving `output/Module.Path/*.js` to use this alias
-/var/www/trypurescript/staging/.psci_modules/.psci_modules/$1
+/var/www/trypurescript/staging/.psci_modules/$1
```

---

**Checklist:**

- [ ] Added the change to the changelog's "Unreleased" section with a reference to this PR (e.g. "- Made a change (#0 by @)")
- [ ] Linked any existing issues or proposals that this pull request should close
- [ ] Updated or added relevant documentation
- [ ] Added a test for the contribution (if applicable)

### CODE REVIEW

**Comment by JordanMartinez** [comment]
I'll have to complete this another day.

**Review by thomashoneyman — APPROVED** [review]

**Review by MonoidMusician — APPROVED** [review]
Looks good. Two minor things:
1. I think `client/setBase.mjs` can be removed now?
2. Don't forget to update the README to _not_ mention `base` now ;)
